samples/bootloader/src/main.c

```diff
@@ -141,12 +154,12 @@ void main(void)
 	const struct fw_info *s0_info = fw_info_find(s0_addr);
 	const struct fw_info *s1_info = fw_info_find(s1_addr);
 
-	if (!s1_info || (s0_info->version >= s1_info->version)) {
-		boot_from(s0_info);
-		boot_from(s1_info);
+	if (!s1_info || (s0_info->version > s1_info->version)) {
+		boot_from(s0_info, 0);
+		boot_from(s1_info, 1);
```

Comment:
We should have defines for the `0` and `1` here to make the code more readable. 

```suggestion
		boot_from(s0_info, BOOT_SLOT_0);
		boot_from(s1_info, BOOT_SLOT_1);
```